src/main/kotlin/org/rust/ide/utils/import/ImportCandidatesCollector.kt

```diff
@@ -61,6 +61,14 @@ object ImportCandidatesCollector {
             // check that result after import can be resolved and resolved element is suitable
             // if no, don't add it in candidate list
             .filter { canBeResolvedToSuitableItem(importingPathText, importContext, it.info) }
+            // make std appear first
+            .sortedBy {
+                if (it.qualifiedNamedItem.containingCrate?.origin == PackageOrigin.STDLIB) {
+                    "0${it.qualifiedNamedItem}"
+                } else {
+                    "1${it.qualifiedNamedItem}"
```

Comment:
The current approach creates a new string object on each comparison. Let's try to avoid it.
You can create a composite comparator to compare `origin` first and only after that compare qualified paths.
`compareBy` and `thenBy` should help you to create the corresponding comparator and `sortedWith` method to use it.
Also, I'd recommend not to use `it.qualifiedNamedItem.toString()` to get qualified name (currently it's used implicitly to construct `"1${it.qualifiedNamedItem}"` string) and use `ImportInfo.usePath` for it

Reply:
As an alternative, I can suggest implementing `Comparable` for `ImportCandidate` class